examples/text_options.rs

```diff
@@ -1,9 +1,17 @@
-use inquire::{error::CustomUserError, length, required, ui::RenderConfig, Text};
+use inquire::{
+    error::CustomUserError,
+    ui::RenderConfig,
+    validator::{ExactLengthValidator, ValueRequiredValidator},
+    Text,
+};
 
 fn main() {
     let answer = Text::new("What's your name?")
         .with_suggester(&suggester)
-        .with_validators(&[required!(), length!(10)])
+        .with_validators(&[
+            Box::new(ValueRequiredValidator::new()),
+            Box::new(ExactLengthValidator::new(10)),
+        ])
```

Comment:
I'm also not a huge fan of this syntax, even with simple macros such as `required!()`, the Box thing adds a huge overhead.

Thought of even removing `with_validators` and pointing users to make multiple calls to `with_validator`, but I think that would be bad for people that define their validators somewhere else and just line them up in this call.

Just wanted to express my thoughts, I understand there's not somewhere to go from here =/

Reply:
Same opinion here.

What do you think about updating all examples to repeatedly call `with_validator` instead? So the API would still exist for people who directly want to pass a pre-created slice, but we show the _preferred_ way in the examples, guiding users to a visually cleaner approach (at the cost of eventual extra allocations due to `Vec` re-sizing, but not a huge concern in my opinion).

Reply:
I like it, please let's do that.

>(at the cost of eventual extra allocations due to Vec re-sizing, but not a huge concern in my opinion)

Say what, let's create the vector with a initial capacity of 5 (or any other value you see reasonable). Something that is small enough for memory usage and large enough for the common use of validators.

Reply:
5 sounds like a good value. After all, it's a `Vec` of `Box`s, so it'll only be 64 bytes allocated for each prompt's validators ([`24 + 5 * 8 = 64`, on 64-bit](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c03436a75072ea226f5883d5384f83f6)).